src/pages/index.js

```diff
@@ -51,36 +85,83 @@ imagePopup.setEventListeners();
 
 const profileFormValidate = new FormValidator("#profile-form", config);
 const cardFormValidate = new FormValidator("#card-form", config);
+const avatarValidate = new FormValidator("#avatar-form", config);
 
 profileFormValidate.enableValidation();
 cardFormValidate.enableValidation();
+avatarValidate.enableValidation();
 
 //inputs and information
 
 // When you click the add card button (the +), it creates a new card.
 function handleCardContent({ place: name, link }) {
-  cardList.addItem(createCard({ name, link }));
-  cardFormValidate.resetForm();
-  newCardModal.close();
+  api.addCard(JSON.stringify({ name, link })).then((card) => {
+    //creating card on server
+    cardList.addItem(createCard({ name, link, id: card._id })); //after getting response, creating card in browser
+    cardFormValidate.resetForm();
+    newCardModal.close();
+  });
 }
 
 function handleImageClick(data) {
-  imagePopup.open(data.name, data.link);
+  imagePopup.open(data._name, data._link);
+}
+
+function handleDeleteConfirm(card) {
+  deleteModal.open(); // opens modal
+  deleteModal.submitAction(() => {
+    api
+      .removeCard(card.id)
+      .then(() => {
+        card.removeCard(); //removes card
+        deleteModal.close();
+      })
+      .catch(() => {});
+  });
 }
 
 function createCard(data) {
-  const card = new Card(data, config.templateSelector, handleImageClick);
+  const card = new Card(
+    data,
+    config.templateSelector,
+    handleImageClick,
+    handleDeleteConfirm
+  );
   return card.getCardElement();
 }
 
+let cards = [];
 const cardList = new Section(
   {
-    items: initialCards,
+    items: cards,
     renderer: (data) => {
       cardList.addItem(createCard(data));
     },
   },
   config.containerSelector
 );
 
+api.loadUserInfo().then((userInfo) => {
+  profileUserInfo.setUserInfo({
+    name: userInfo.name,
+    description: userInfo.about,
+    avatar: userInfo.avatar,
+  });
+});
+
+api.getInitialCards().then((result) => {
+  result.reverse().forEach((item) => {
+    cardList.addItem(
+      createCard({
+        name: item.name,
+        link: item.link,
+        id: item._id,
+        isLiked: item.isLiked,
+      })
+    );
+  });
+});
+
 cardList.renderItems();
```

Comment:
It doesn't work here. You need to get the cards from the server and then render them